### PR TITLE
Move login function to commands.ts file

### PR DESCRIPTION
**Pull request by campionfellin**
This PR does a couple of things:

* Moves `login` functionality to `commands.ts` file. 
* Also adds `process.exit(0)` after `clasp login` so that it will end after logging in (otherwise after you login you need `CTRL+C` after logging in).

Signed-off-by: campionfellin <campionfellin@gmail.com>

Works towards a model for #134 

- [x] `npm run test` succeeds.
- [x] `npm run lint` succeeds.
- [ ] Appropriate changes to README are included in PR.

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -168,17 +168,7 @@ commander
...
-      console.warn(ERROR.LOGGED_IN);
-    }).catch(async (err: string) => {
-      DOTFILE.RC_LOCAL.read().then((rc: ClaspSettings) => {
-        console.warn(ERROR.LOGGED_IN);
-      }).catch(async (err: string) => {
-        await checkIfOnline();
-        authorize(options.localhost, options.ownkey);
-      });
-    });
+    login(options);
```
Comment (grant) [inline index.ts]: If you look at this, we're calling a function with `options` that just calls a function with `options`.

I bet you can just do this:
```ts
commander
  .command('login')
  .description('Log in to script.google.com')
  .option('--no-localhost', 'Do not run a local server, manually enter code instead')
  .option('--ownkey', 'Save .clasprc.json file to current working directory')
  .action(login);
```

File: src/auth.ts [hunk 2]
```diff
@@ -115,6 +115,7 @@ export async function authorize(useLocalhost: boolean, writeToOwnKey: boolean) {
     const token = await (useLocalhost ? authorizeWithLocalhost() : authorizeWithoutLocalhost());
     await (writeToOwnKey ? DOTFILE.RC_LOCAL.write(token) : DOTFILE.RC.write(token));
     console.log(LOG.AUTH_SUCCESSFUL);
+    process.exit(0);
```
Comment (grant) [inline src/auth.ts]: A comment as to why this is needed wouldn't hurt.

File: src/commands.ts [hunk 3]
```diff
@@ -37,3 +39,16 @@ export const LOG = {
         (description || '(no description)'),
     VERSION_NUM: (numVersions: number) => `~ ${numVersions} ${pluralize('Version', numVersions)} ~`,
   };
+
+export function login(options) {
+    DOTFILE.RC.read().then((rc: ClaspSettings) => {
```
Comment (grant) [inline src/commands.ts]: Format indentation.

File: src/commands.ts [hunk 4]
```diff
@@ -37,3 +39,16 @@ export const LOG = {
         (description || '(no description)'),
     VERSION_NUM: (numVersions: number) => `~ ${numVersions} ${pluralize('Version', numVersions)} ~`,
   };
+
+export function login(options) {
```
Comment (grant) [inline src/commands.ts]: Don't remove the param types.
```ts
export function login(options:{ localhost: boolean, ... }) {
```

File: src/commands.ts [hunk 5]
```diff
@@ -37,3 +39,16 @@ export const LOG = {
         (description || '(no description)'),
     VERSION_NUM: (numVersions: number) => `~ ${numVersions} ${pluralize('Version', numVersions)} ~`,
   };
+
```
Comment (grant) [inline src/commands.ts]: Add a JSDoc comment.

**Comment by campionfellin** [comment]
Will address those changes in a second here. Can you investigate why the travis build is failing? It seems like it only fails on Node (without specific version, maybe node 10?)

**Comment by campionfellin** [comment]
This is why we need the `process.exit(0);`


clasp-login-prob

**Comment by grant** [comment]
Re: The `process.exit`, yeah I know why it's added, but add a comment in the code. Also, is this in prod and should be patched ASAP?

Re: Travis failure
Looks like `grpc` uses `node-pre-gyp` which fails for some reason on the latest version of Node:

```
node-pre-gyp ERR! Tried to download(403): https://storage.googleapis.com/grpc-precompiled-binaries/node/grpc/v1.10.1/node-v64-linux-x64-glibc.tar.gz 
```

I don't know why this is though. Maybe try changing the grpc version or `package-lock.json`?

**Comment by campionfellin** [comment]
Yes, looks like it was introduced here: https://github.com/google/clasp/pull/177/files

Scroll to bottom, line 117 of `src/auth.ts` (https://github.com/google/clasp/blob/6dc109bdd532efd761e2e6ae5a415041dcc39cd4/src/auth.ts#L117).

I'll finish this PR in a second here and we can merge it in.

File: src/commands.ts (hunk 5 above)
Comment (campionfellin) [reply src/commands.ts]: 👍

File: src/auth.ts (hunk 2 above)
Comment (campionfellin) [reply src/auth.ts]: 👍

**Comment by ajlai** [comment]
Yikes, looks like hanging is from http_server.close() taking longer to gracefuly shut down. It looks like it takes ~4-5 seconds for me after authorizing for the clasp login command to close.

**Comment by grant** [comment]
Had to restart the Travis build to get `grpc` to work.
For some reason it was still doing this `grpc@1.10.1 install /home/travis/build/google/clasp/node_modules/grpc`.

**Pull request merged**